editoast/src/views/v2/train_schedule.rs

```diff
@@ -255,37 +267,47 @@ async fn put(
     Ok(Json(ts_result.into()))
 }
 
-#[derive(Deserialize, Serialize, Clone, Debug, ToSchema)]
-struct SimulationOutput {
-    base: ReportTrain,
-    provisional: ReportTrain,
-    final_output: CompleteReportTrain,
-    mrsp: Mrsp,
-    power_restriction: String,
+// TODO: When can the simulation fail ?
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+#[serde(tag = "status", rename_all = "snake_case")]
+#[allow(clippy::large_enum_variant)]
+enum SimulationResult {
+    #[allow(clippy::large_enum_variant)]
```

Comment:
Maybe give that `#[allow(...)]` a comment.

Suggestion: The other variant is only used when an error occurs, it's preferable to avoid indirection from a `Box`on a happy path, and it's ok to waste some memory in an error path.